pkg/destroy/ovirt/destroyer.go

```diff
@@ -53,20 +56,34 @@ func (uninstaller *ClusterUninstaller) removeVMs(con *ovirtsdk.Connection) error
 	// - stop + delete VMS
 	vms := vmsResponse.MustVms().Slice()
 	uninstaller.Logger.Debugf("Found %d VMs", len(vms))
+	errChan := make(chan (error), len(vms))
 	wg := sync.WaitGroup{}
 	wg.Add(len(vms))
 	for _, vm := range vms {
 		go func(vm *ovirtsdk.Vm) {
-			uninstaller.stopVM(vmsService, vm)
-			uninstaller.removeVM(vmsService, vm)
-			wg.Done()
+			defer wg.Done()
+			if err := uninstaller.stopVM(ctx, vmsService, vm); err != nil {
+				errChan <- err
+				return
+			}
+			if err := uninstaller.removeVM(vmsService, vm); err != nil {
+				errChan <- err
+			}
 		}(vm)
 	}
 	wg.Wait()
-	return nil
+	select {
+	case err := <-errChan:
+		return err
+	default:
+		return nil
+	}
```

Comment:
why not collection all the errors instead of first one?

Reply:
I changed this to aggregate all of the errors.